mlfinlab/microstructural_features/third_generation.py

```diff
@@ -14,5 +14,7 @@ def get_vpin(volume: pd.Series, buy_volume: pd.Series, window: int = 1) -> pd.Se
     :return: (pd.Series) VPIN series
     """
     sell_volume = volume - buy_volume
-    volume_imbalance = abs(buy_volume - sell_volume)
-    return volume_imbalance.rolling(window=window).mean() / volume
```

Comment:
This does not work well for bar with different volume sizes and window>1, which can make VPIN goes above 1. 

Instead, return volume_imbalance.rolling(window=window).sum() / volume.rolling(window=window).sum() works well for both cases

Reply:
well... the name VPIN -> "volume" pin

Reply:
Don't get me wrong, the implemented code should be right if all bars have the same volume. However, extract the volume bars from 1min time bars could lead to peaks of volume bars above the threshold, specially next toe market open and close. The code I pasted should fix this and also limit VPIN to [0,1] even if the bars have different volume sizes. Also, both codes should have the same outcome for same size volume bars.

Reply:
i agree with you =)

Reply:
Awesome!!